packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1953,289 +1953,282 @@ async function spawnDynamicValidationInDev(
 ): Promise<void> {
   const { componentMod: ComponentMod } = ctx
 
+  // Prerender controller represents the lifetime of the prerender.
+  // It will be aborted when a Task is complete or a synchronously aborting
+  // API is called. Notably during cache-filling renders this does not actually
+  // terminate the render itself which will continue until all caches are filled
+  const initialServerPrerenderController = new AbortController()
```

Comment:
are these codepaths similar enough that would be hoisting into a shared function to avoid validation drift or is there still meaningful difference between them?

Reply:
yes but that's coming a little later in the refactor